modules/monitoring-configuring-remote-write.adoc

```diff
@@ -0,0 +1,211 @@
+// Module included in the following assemblies:
+//
+// * monitoring/configuring-the-monitoring-stack.adoc
+
+[id="configuring_remote_write_storage_{context}"]
+= Configuring remote write storage
+
+You can configure remote write to enable Prometheus to send ingested metrics to remote systems for long-term storage. 
+Doing so has no impact on how or for how long Prometheus stores metrics.
+
+.Prerequisites
+
+* *If you are configuring core {product-title} monitoring components:*
+** You have access to the cluster as a user with the `cluster-admin` role.
+** You have created the `cluster-monitoring-config` `ConfigMap` object.
+* *If you are configuring components that monitor user-defined projects:*
+** You have access to the cluster as a user with the `cluster-admin` role or as a user with the `user-workload-monitoring-config-edit` role in the `openshift-user-workload-monitoring` project.
+** You have created the `user-workload-monitoring-config` `ConfigMap` object.
+* You have installed the OpenShift CLI (`oc`).
+* You have set up a remote write compatible endpoint (such as Thanos) and know the endpoint URL.
+See the link:https://prometheus.io/docs/operating/integrations/#remote-endpoints-and-storage[Prometheus remote endpoints and storage documentation] for information about endpoints that are compatible with the remote write feature.
+* You have set up authentication credentials for the remote write endpoint. 
++
+[CAUTION]
+====
+To reduce security risks, avoid sending metrics to an endpoint via unencrypted HTTP or without using authentication.
+====
+
+.Procedure
+
+. Edit the `cluster-monitoring-config` `ConfigMap` object in the `openshift-monitoring` project:
++
+[source,terminal]
+----
+$ oc -n openshift-monitoring edit configmap cluster-monitoring-config
+----
+
+. Add a `remoteWrite:` section under `data/config.yaml/prometheusK8s`.
+
+. Add an endpoint URL and authentication credentials in this section:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://remote-write.endpoint"
+        <endpoint_authentication_credentials>
+----
++
+For `endpoint_authentication_credentials` substitute the endpoint's credentials.
+Currently supported authentication methods are basic authentication (`basicAuth`) and client TLS (`tlsConfig`) authentication.
++
+* The following example configures basic authentication:
++
+[source,yaml]
+----
+basicAuth:
+  username:
+    <usernameSecret>
+  password:
+    <passwordSecret>
+----
+Substitute `<usernameSecret>` and `<passwordSecret>` accordingly. The following sample shows basic authentication configured with `remoteWriteAuth` for the `name` values and `user` and `password` for the `key` values, which contain the endpoint authentication credentials:
```

Comment:
The previous sentence is about the preceding sample. Suggest starting the 'The following sample...' sentence on a new paragraph to indicate the switch in the context.
Also, suggest breaking the second sentence into two sentences.

Reply:
@Preeticp: Revised per your suggestion